File: frontend/app/types/certificate.ts
export interface Certificate {
  id: string;
  subject: string;
  issuer: string;
  expiration: string;
  san_entries: string[];
  created_at: string;
}

export interface CertificateListResponse {
  data: Certificate[];
  next_cursor: string | null;
  has_more: boolean;
  total: number;
  expiring_soon_count: number;
}

File: backend/src/domain/certificate.rs
use chrono::{DateTime, Utc};
use uuid::Uuid;

// Core domain entity
pub struct Certificate {
    pub id: Uuid,
    pub subject: String,
    pub issuer: String,
    pub expiration: DateTime<Utc>,
    pub san_entries: Vec<String>,
    pub created_at: DateTime<Utc>,
}
